src/components/sections/items/GenericSectionItem.tsx

```diff
@@ -1,20 +1,21 @@
 import React, {Children, PropsWithChildren} from 'react';
-import {AccessibilityProps, View} from 'react-native';
+import {AccessibilityProps, StyleProp, View, ViewStyle} from 'react-native';
 import {useSectionItem} from '../use-section-item';
 import {SectionItemProps} from '../types';
 import {useSectionStyle} from '../use-section-style';
 
 type Props = PropsWithChildren<
   SectionItemProps<{
     accessibility?: AccessibilityProps;
+    style?: StyleProp<ViewStyle>;
```

Comment:
I feel like it is a little can of worms to expose the internal `style` of a section item. Could consider using a more specific field, like `shouldGrow: boolean` or something here.

Reply:
Yes, I was considering something like that myself, but wasn't sure what I felt was the best option. I believe a prop like `shouldGrow` is kinda similar to `withPadding`, in that you use a prop to dictate appearance. In that sense i like the `style` approach better. But with increased freedom also comes increased responsibility. It needs to be used with caution.

Reply:
@gorandalum What do you think?

Reply:
Not meaning to delay the merge on such minor details 😅 

My preference still goes towards `shouldGrow`. The reason for moving away from `withPadding` and such was that we saw that they were too limiting, as `Section` can't foresee every usage from the ouside (thus over time also adding `withFullPadding`, `withTopPadding`, `withBottomPadding`). In this case it is internally in the `Section`, and then we generally want to be as limiting as possible.

Reply:
If we come back to this prop in a year I think it will be a bit hard to grasp what `shouldGrow` is actually doing. As you say, to not stall this pr any longer I'll just merge it with the `style` prop. Personally I don't think that will case any major issues. Most devs in our are reasonably responsible, and the code review process should probably pick up the rest.